cpp/epidemiology/utils/eigen_util.h

```diff
@@ -255,6 +257,23 @@ class RowMajorIterator
     {
     }
 
+    /**
+     * Default copy constructor and copy assignment operator, default move constructor and move assignment operator
+     */
+    RowMajorIterator(const RowMajorIterator& other)
+        : m_matrix(other.m_matrix)
+        , m_i(other.m_i)
+    {
+    }
+    RowMajorIterator& operator=(const RowMajorIterator& other)
+    {
+        m_matrix = other.m_matrix;
```

Comment:
this is a bug. not your change, that just uncovered it. 

this assignment doesn't rebind the reference but overwrites the contents of the referenced matrix. The iterator needs to store a shallow reference, like a pointer or reference_wrapper. then the implicit constructor/operator works.